authentication_sample.js

```diff
@@ -30,35 +32,20 @@ const secret = process.env['AZURE_CLIENT_SECRET'];
 // Random sample keyvault name
 const kvName = random_id();
 
-// Authenticates to the Azure Key Vault by providing a callback to authenticate using ADAL.
-function authUsingAdalCallback(vaultUri) {
-    console.log("Using ADAL to authenticate to '" + vaultUri + "'");
+function authUsingAAD(vaultUri) {
+    console.log("Using AAD to authenticate to '" + vaultUri + "'");
     
-    // Callback for ADAL authentication.
-    const adalCallback = (challenge, callback) => {
-        const context = new AuthenticationContext(challenge.authorization);
-        return context.acquireTokenWithClientCredentials(challenge.resource, clientId, secret, (err, tokenResponse) => {
-            if(err) {
-                throw err;
-            }
-            
-            // The KeyVaultCredentials callback expects an error, if any, as the first parameter. 
-            // It then expects a value for the HTTP 'Authorization' header, which we compute based upon the access token obtained with the SP client credentials. 
-            // The token type will generally equal 'Bearer' - in some user-specific situations, a different type of token may be issued. 
-            return callback(null, tokenResponse.tokenType + ' ' + tokenResponse.accessToken);
-        });
-    };
-    
-    const keyVaultClient = new KeyVault.KeyVaultClient(new KeyVault.KeyVaultCredentials(adalCallback));
+    const credentials = new DefaultAzureCredential();
```

Comment:
Since the other half of the sample is using the service principal secret way of doing auth (msRestNodeAuth.loginWithServicePrincipalSecret), we should do the same here i.e. use the `ClientSecretCredential` class from `@azure/identity`. 

In a month or two, all the management plane packages will be updated to support `@azure/identity` as well. Then, we can come back here and update the sample to not use ms-rest-nodeauth anymore

Reply:
OK, it's updated to use the ClientSecretCredential now.

 I will update it when the management plane packages has been updated.